drivers/src/pmu.rs

```diff
@@ -0,0 +1,117 @@
+// Copyright (c) 2022 by Rivos Inc.
+// Licensed under the Apache License, Version 2.0, see LICENSE for details.
+// SPDX-License-Identifier: Apache-2.0
+
+use arrayvec::ArrayVec;
+use sbi::{api::pmu, Error as SbiError, PmuCounterInfo, Result as SbiResult};
+use spin::Once;
+
+/// Maximum number of supported platform PMU counters
+pub const MAX_HARDWARE_COUNTERS: usize = 32;
+/// Caches information about platform hardware counters.
+#[derive(Debug)]
+pub struct PmuInfo {
+    counters_info: ArrayVec<Option<PmuCounterInfo>, MAX_HARDWARE_COUNTERS>,
+    counters_mask: u64,
```

Comment:
Can you rename it to hardware_counter_mask or something similar? It's bit confusing while grepping the code as all SBI function arguments also use counters_mask.